### PR TITLE
Filter rayo module

### PR DESCRIPTION
**Pull request by damencho**
Adds a prosody module to filter rayo incoming iqs if user is not authenticated using jwt. Hides the button in the UI if roles evaluation is enabled in config and the user is guest.

Depends on https://github.com/jitsi/jitsi-meet/pull/1548

### CODE REVIEW

File: resources/prosody-plugins/mod_filter_iq_rayo.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+-- filters rayo iq in case of requested from not jwt authenticated sessions
+module:hook("pre-iq/full", function(event)
+    local stanza = event.stanza;
+    if stanza.name == "iq" then
+        local dial = stanza:get_child('dial', 'urn:xmpp:rayo:1');
+        if dial then
+            local session = event.origin;
+            local token = session.auth_token;
+            if token == nil then
```
Comment (paweldomas) [inline resources/prosody-plugins/mod_filter_iq_rayo.lua]: I wonder if room name should be validated here as well ?

**Comment by damencho** [comment]
Pushed some changes, which depends on https://github.com/jitsi/jitsi-meet/pull/1548. They now use the room name from the iq to validate it using token_util.

File: resources/prosody-plugins/mod_filter_iq_rayo.lua [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    local stanza = event.stanza;
+    if stanza.name == "iq" then
+        local dial = stanza:get_child('dial', 'urn:xmpp:rayo:1');
+        if dial then
+            local session = event.origin;
+            local token = session.auth_token;
+            local header = dial:get_child('header');
+
+            if token == nil
+                or header.attr.name ~= 'JvbRoomName'
```
Comment (paweldomas) [inline resources/prosody-plugins/mod_filter_iq_rayo.lua]: what if there are more headers ? does `dial:get_child('header')` return the first child element ?

Comment (damencho) [inline resources/prosody-plugins/mod_filter_iq_rayo.lua]: Fixed it.

Comment (paweldomas) [inline resources/prosody-plugins/mod_filter_iq_rayo.lua]: wow it's so complex to write it in lua :P

**Pull request merged**